discretisedfield/field.py

```diff
@@ -3790,27 +3790,36 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         result = getattr(ufunc, method)(*inputs, **kwargs)
         if isinstance(result, tuple):
             if len(result) != len(mesh):
-                raise ValueError("wrong number of Field objects")
-            return tuple(
-                self.__class__(
-                    m,
-                    nvdim=x.shape[-1],
-                    value=x,
-                    vdims=self.vdims,
-                    vdim_mapping=self.vdim_mapping,
+                raise NotImplementedError()
+            try:
+                return tuple(
+                    self.__class__(
+                        m,
+                        nvdim=x.shape[-1],
+                        value=x,
+                        vdims=self.vdims,
+                        vdim_mapping=self.vdim_mapping,
+                    )
+                    for x, m in zip(result, mesh)
                 )
-                for x, m in zip(result, mesh)
-            )
+            except Exception:
+                raise NotImplementedError()
         elif method == "at":
             return None
         else:
-            return self.__class__(
-                mesh[0],
-                nvdim=result.shape[-1],
-                value=result,
-                vdims=self.vdims,
-                vdim_mapping=self.vdim_mapping,
-            )
+            print("Hi")
+            if not np.array_equal(result.shape[:-1], mesh[0].n):
+                raise NotImplementedError()
+            try:
+                return self.__class__(
+                    mesh[0],
```

Comment:
Here as well, can we do this?
```suggestion
                    self.mesh,
```